scripts/files/file_tiff.py

```diff
@@ -96,22 +122,34 @@ def check_no_data(self, gdalinfo: GdalInfo) -> None:
                     custom_fields={"current": f"{current_nodata_val}"},
                 )
         else:
-            self.add_error(error_type=FileTiffErrorType.NO_DATA, error_message="noDataValue not set")
+            if len(bands) == 4 and bands[3]["colorInterpretation"] != "Alpha":
+                self.add_error(error_type=FileTiffErrorType.NO_DATA, error_message="noDataValue not set")
+
+    def is_no_data(self, gdalinfo: GdalInfo) -> bool:
```

Comment:
You can make `gdalinfo` as optional (for testing?) and call `get_gdalinfo_original()` if `_gdalinfo_original` is None